runners/portability/java/src/main/java/org/apache/beam/runners/portability/ExternalWorkerService.java

```diff
@@ -90,10 +97,44 @@ public void stopWorker(
   public void close() {}
 
   public GrpcFnServer<ExternalWorkerService> start() throws Exception {
-    GrpcFnServer<ExternalWorkerService> server =
-        GrpcFnServer.allocatePortAndCreateFor(this, serverFactory);
+    final String externalServiceAddress =
+        Environments.getExternalServiceAddress(options.as(PortablePipelineOptions.class));
+    GrpcFnServer<ExternalWorkerService> server;
+    if (externalServiceAddress.isEmpty()) {
+      server = GrpcFnServer.allocatePortAndCreateFor(this, serverFactory);
+    } else {
+      server =
+          GrpcFnServer.create(
+              this,
+              Endpoints.ApiServiceDescriptor.newBuilder().setUrl(externalServiceAddress).build(),
+              serverFactory);
+    }
     LOG.debug(
         "Listening for worker start requests at {}.", server.getApiServiceDescriptor().getUrl());
     return server;
   }
+
+  public static void main(String[] args) throws Exception {
```

Comment:
Perhaps documentation with an example how to start the worker pool would be useful?

Reply:
Got you, I will update the javadoc.

In addition, I am planning to have a separate PR out to update the `java/container/boot.go` to accept `worker_pool` option, just like `python/container/boot.go` so that `boot.go` can start `ExternalWorkerService` instead of `FnHarness` like `java -cp xxx org.apache.beam.runners.portability.ExternalWorkerService` 

Do you think this aligns your thoughts on how java should support worker pool ?

Reply:
Yes, that makes sense. From a user's perspective it is preferable to keep the container contract uniform.

Reply:
Updated